src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -266,6 +266,33 @@ internal static unsafe string AnsiBstrBufferToString(byte* buffer)
             return Marshal.PtrToStringAnsi((IntPtr)buffer, (int)Marshal.SysStringByteLen((IntPtr)buffer));
         }
 
+        internal static unsafe decimal NativeToManagedDecimal(NativeDecimal* value)
+        {
+            byte* pValue = (byte*)value;
+
+            var signscale = *(ushort*)(pValue + 2);
+            var hi32 = *(uint*)(pValue + 4);
+            var lo64 = *(ulong*)(pValue + 8);
+
+            var sign = (byte)((signscale & 0xFF00) >> 8);
+            var scale = (byte)(signscale & 0xFF);
+            var mid32 = (int)((lo64 & 0xFFFFFFFF00000000) >> 32);
+            var lo32 = (int)(lo64 & 0xFFFFFFFF);
+
+            return new decimal(lo32, mid32, (int)hi32, sign != 0, scale);
```

Comment:
Calling this constructor will do some extra argument validation for `scale`. I do not see it in the CoreCLR decimal marshalers. CoreCLR marshalers just blasts the bits over. It would be nice to match what CoreCLR does.